### ISSUE TITLE
Consider new "raw" option for horizontal axis

### ISSUE THREAD

**Issue by bersbersbers** [tensorflow/tensorboard#2025]
This issue is motivated by an issue in current TensorFlow, which incorrectly annotates the `step` parameter for `epoch_` summaries when using `update_freq` (https://github.com/tensorflow/tensorflow/issues/26873).

Anyway, I have a number of TF summaries that are impossible to compare because the x-axis values are meaningless. I can think of other examples where this may happen, for example, when using training sets of different sizes (let's say 100 and 1000 samples) and using different values of `update_freq` (e.g., `=10` for the first and  `=100` for the latter to get the same number of within-batch metrics saved.

The current x-axis options (step, relative [time], wall [time]) are not appropriate in these situations. Please consider an additional "raw" option that basically ignores the step values from the summary files.

**Comment by hgadig** [comment]
@nfelt  Any update on this ?

**Comment by nfelt** [comment]
Thanks for the feature request.  Just so that I understand, by "raw" do you mean the index/offset of the summary in the file?  E.g. just laying them out linearly as 1, 2, 3, 4, 5, ... regardless of the actual step?

This is a little tricky to do correctly if the same tag is written across multiple files, because it's not necessarily clear how to order data from the two files when they're interleaved.  For your use case, would it hypothetically work to first sort all the summary values according to wall time, and then take their index values from the combined list and use those on the x-axis?

Lastly, is it sufficient for your use case to do this dashboard-wide using the existing x-axis setting, or would you have any preference to be able to set the x-axis for individual charts separately?

**Comment by bersbersbers** [comment]
Thanks a lot for considering this!

> Thanks for the feature request. Just so that I understand, by "raw" do you mean the index/offset of the summary in the file? E.g. just laying them out linearly as 1, 2, 3, 4, 5, ... regardless of the actual step?

Yes, correct!

> This is a little tricky to do correctly if the same tag is written across multiple files, because it's not necessarily clear how to order data from the two files when they're interleaved. For your use case, would it hypothetically work to first sort all the summary values according to wall time, and then take their index values from the combined list and use those on the x-axis?

I cannot say for sure since I try to avoid splitting runs across multiple files (I am not even aware I ever used this). But your proposed solution is the first thing that came to my mind: Sort like you normally would, then replace steps by subsequent numbers. (I *am* wondering whether it makes a difference to sort by wall time or anything else first, but I cannot think of any use case of mine where that would make a difference.)

> Lastly, is it sufficient for your use case to do this dashboard-wide using the existing x-axis setting, or would you have any preference to be able to set the x-axis for individual charts separately?

Me, personally, I would be fine with a dashboard-wide setting.
